services/cd-service/pkg/repository/transformer.go

```diff
@@ -374,9 +373,79 @@ func (c *CreateApplicationVersion) Transform(ctx context.Context, state *State)
 			result = result + deployResult + "\n"
 		}
 	}
+	err = writeCommitData(ctx, c.SourceCommitId, c.Application, c.EventId, allEnvsOfThisApp, fs)
+	if err != nil {
+		return "", nil, GetCreateReleaseGeneralFailure(err)
+	}
+
 	return fmt.Sprintf("created version %d of %q\n%s", version, c.Application, result), changes, nil
 }
 
+func writeCommitData(ctx context.Context, sourceCommitId string, app string, eventId string, environments []string, fs billy.Filesystem) error {
+	if !valid.SHA1CommitID(sourceCommitId) {
+		return nil
+	}
+	commitDir := commitApplicationDirectory(fs, sourceCommitId, app)
+	if err := fs.MkdirAll(commitDir, 0777); err != nil {
+		return err
+	}
+	if err := util.WriteFile(fs, fs.Join(commitDir, ".dir"), make([]byte, 0), 0666); err != nil {
+		return err
+	}
+	err := writeEvent(ctx, eventId, sourceCommitId, fs, environments)
+	if err != nil {
+		return fmt.Errorf("error while writing event: %v", err)
+	}
+
+	return nil
+}
+
+func writeEvent(ctx context.Context, eventId string, sourceCommitId string, filesystem billy.Filesystem, envs []string) error {
+	if eventId == "" {
+		fullUuid, err := uuid.NewUUID()
+		if err != nil {
+			return fmt.Errorf("could not create uuid: %v", err)
+		}
+		// format of this "uuid" is the same as the lock IDs, see "randomLockId" in store.tsx
+		eventId = "cd-v1-" + string(fullUuid.String()[:7])
+	}
+	eventDir := commitEventDir(filesystem, sourceCommitId, eventId)
+	_, err := filesystem.Stat(eventDir)
+	if err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			// anything except "not exists" is an error:
+			return fmt.Errorf("event directory %s already existed: %v", eventDir, err)
+		}
+	}
+	if err := filesystem.MkdirAll(eventDir, 0777); err != nil {
+		return fmt.Errorf("could not create directory %s: %v", eventDir, err)
+	}
+	for i := range envs {
+		env := envs[i]
+		environmentDir := filesystem.Join(eventDir, "environments", env)
+		if err := filesystem.MkdirAll(environmentDir, 0777); err != nil {
+			return fmt.Errorf("could not create directory %s: %v", environmentDir, err)
+		}
+		environmentNamePath := filesystem.Join(environmentDir, ".dir")
+		if err := util.WriteFile(filesystem, environmentNamePath, []byte(""), 0666); err != nil {
+			return fmt.Errorf("could not write file %s: %v", environmentNamePath, err)
+		}
+	}
+	const (
+		EventTypeNewRelease = "new-release"
+	)
+	eventTypePath := filesystem.Join(eventDir, "eventType")
+	if err := util.WriteFile(filesystem, eventTypePath, []byte(EventTypeNewRelease), 0666); err != nil {
+		return fmt.Errorf("could not write file %s: %v", eventTypePath, err)
+	}
+
+	createdAtPath := filesystem.Join(eventDir, "createdAt")
```

Comment:
I we would use uuids as event ids, we could just get this from the event id.

Reply:
This would require us to use a certain kind of uuid. I think it's simpler to just write the file.

Reply:
This depends on the UUID implementation, so I'd prefer to have the data explicitely.

Reply:
This is not accurate. UUIDv1 is an independant spec that is not depending on any implementation

Reply:
as discussed, we are now using time based UUIDs in this PR